.github/workflows/ci.yml

```diff
@@ -0,0 +1,70 @@
+name: CI
+
+on:
+  workflow_dispatch: # Can manually run from any branch
+  push:
+    branches:
+      - trunk
+  pull_request:
+
+jobs:
+  php-tests:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        php: ['8.1', '8.0', '7.4']
+        os: ['ubuntu-latest']
+        include:
+          - php: '7.3'
+            os: 'ubuntu-18.04' # PHP 7.3 is not pre-installed in ubuntu-latest
+      fail-fast: false
+    env:
+      COVERAGE_CACHE_PATH: phpunit-coverage-cache
+      PHP_VERSION: ${{ matrix.php }}
+
+    steps:
+      - uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # v2.4.2
+
+      - name: Switch to PHP ${{ env.PHP_VERSION }}
+        run: |
+          sudo update-alternatives --set php /usr/bin/php$PHP_VERSION
+          sudo update-alternatives --set phar /usr/bin/phar$PHP_VERSION
+          sudo update-alternatives --set phar.phar /usr/bin/phar.phar$PHP_VERSION
+          sudo update-alternatives --set phpize /usr/bin/phpize$PHP_VERSION
+          sudo update-alternatives --set php-config /usr/bin/php-config$PHP_VERSION
```

Comment:
I think we could use https://github.com/marketplace/actions/setup-php-action to make this cleaner.

Reply:
as discusses with @llessa-godaddy we don't need to refactor that now, however we added a code coverage badge which may be handy